### ISSUE TITLE
[FEATURE] `max_execution_time` config

### ISSUE THREAD

**Issue by rvolosatovs** [wasmCloud/wasmCloud#2016]
### Affected project(s)

- [ ] documentation
- [ ] examples
- [X] wasmCloud host
- [X] wasmCloud CLI (wash)
- [ ] wasmCloud dashboard UI (washboard)
- [ ] capability providers
- [ ] provider bindgen
- [X] control interface client
- [ ] other / not sure

### Is your feature request related to a problem? Please describe.

In #2002 and #2015 support for `max_execution_time` was added to the host and runtime, it should be configurable per-host and also per-component

### Describe the solution you'd like

Host-level config via flags and env vars
Component-level config via control interface

**Comment by brooksmtownsend** [comment]
This is a great first issue to get into 1.1!

**Comment by Aditya1404Sal** [comment]
I'd love to work on this feature, @vados-cosmonic  can I pick this up ?

**Comment by vados-cosmonic** [comment]
Hey I think so @Aditya1404Sal -- thanks for hopping on this!

**Comment by Aditya1404Sal** [comment]
> Hey I think so @Aditya1404Sal -- thanks for hopping on this! IMO you could do it in two commits -- first the host configuration and then the component-level config separately!

Yes that sounds great -- that way it'll be systematic and easily trackable :+1:

**Comment by Aditya1404Sal** [comment]
> Hey I think so @Aditya1404Sal -- thanks for hopping on this! IMO you could do it in two commits -- first the host configuration and then the component-level config separately!

also, could I have this ticket assigned to me? to avoid any conflicts

**Comment by vados-cosmonic** [comment]
Sure!

**Comment by vados-cosmonic** [comment]
Hey @Aditya1404Sal it's probably a good idea to break this up into two distinct PRs -- let's update the host first and control interface/component-level configuration can be handled later!

**Comment by Aditya1404Sal** [comment]
I understand :+1: -- I'll make a separate PR for the Component-level configuration after host is updated :)

**Comment by Aditya1404Sal** [comment]
> Hey @rvolosatovs  -- The Issue ticket suggests adding Component-level configurability for max_execution_time via control interface
> If I'm not wrong , the control interface is a set of API's that can be called from a Client side that can configure the state of a Component/Provider runtime declaratively , moreover these api's are indirectly invoked whenever `wash start component <args> ` or `wash scale` are called. While I understand that functions like  https://github.com/wasmCloud/wasmCloud/blob/bfa9eb5a3380646ecda3ea83903b33970b24f7db/crates/control-interface/src/client.rs#L261 can be configured to incorporate `max_execution_time` , I think there is also a need to add a flag to `wash start component <args>` or a parse field for the context file to enable configuring  it. 
> 
> I need a bit more clarity on what exactly needs to be done regarding the 2nd objective.

@vados-cosmonic , Since the first PR got merged can we get back to this ?

**Comment by vados-cosmonic** [comment]
Hey @Aditya1404Sal the second part is considerably more involved! So you're right that there is the control interface to update, and also the host as well. There is also some async stuff to worry about!

https://github.com/wasmCloud/wasmCloud/issues/2530 is related and maybe a good place to start!

**Comment by Aditya1404Sal** [comment]
Great! I will take a look at #2530 and see what I can do.
Thanks for the clarification @vados-cosmonic !

**Comment by vados-cosmonic** [comment]
Appreciate it @Aditya1404Sal !
